Assets/HoloToolkit/Input/Scripts/InputEventData/BaseInputEventData.cs

```diff
@@ -25,16 +40,28 @@ public abstract class BaseInputEventData : BaseEventData, IInputSourceInfoProvid
         /// </summary>
         public object Tag { get; private set; }
 
+        /// <summary>
+        /// An optional, input-source-independent tag to be associated with this event.
+        /// </summary>
+        public string EventOrigin { get; private set; }
+
+        /// <summary>
+        /// An optional, input-source-dependent tag to be associated with this input source
+        /// </summary>
+        public MixedRealityInputType SourceType { get; private set; }
```

Comment:
Isn't the source kind part of the `IInputSource`?

Should we just get it from there?

Reply:
Probably. See my notes above.

>my feeling is we should move this information into IInputSource after all.